client/src/components/Nav.jsx

```diff
@@ -19,6 +21,12 @@ const Nav = ({ showSidebar, setShowSidebar }) => {
   const DropDownHandler = (showDropDown) => {
     setShowDropDown(!showDropDown);
   };
+  useEffect(() => {
+    if (IsMatch.pathname === "/") setTitle("Today");
+    if (IsMatch.pathname === "/goals") setTitle("Goals");
+    if (IsMatch.pathname === "/shares") setTitle("Shares");
+    if (IsMatch.pathname === "/leaderboard") setTitle("Leaderboard");
+  }, [IsMatch]);
 
   const Logout = () => {
     axios
```

Comment:
Logout route is not working properly. Maybe can try using link if history.push not work. Idk